Create an updated landing page

lokeshyerneni opened the pull request:
[WIP] This is a webpage created for the users to look at when they first enter the Planner website. The inspiration for this was taken from the landing page designed on [Figma](https://www.figma.com/file/cycV0bxP6vTJJYUONFoBr0/Planner-Mockups?node-id=24929%3A31445) in the Main Pages section. As of now, the webpage currently has a static layer with minimal responsiveness and good animation. Lastly, pay no attention to the website when the browser is resized as it will be addressed in the near future when the animations and webpage structure are taken care of. Any feedback regarding animations such as excess animation/lacking animations as well as the structure of the webpage will be appreciated.

lokeshyerneni committed:
feat(component): Create a popup dialog

Create a dialog popup for the users to input information for their personalized
degree plan

fix #91

lokeshyerneni committed:
feat(landing page): Create a landing page for the users

The landing page will be the first webpage that is shown to the users before
anything else takes place

fix #109

lokeshyerneni committed:
feat(landing page): Implement animations onto the landing page

ZzRanger commented:
Please resolve build issues so that we can preview the deployment on Vercel. You can do this by running "npm run build" on your local machine, then resolving all the errors that appear.

ZzRanger commented on components/home/Onboarding/ClassCard.tsx:
```diff
@@ -0,0 +1,140 @@
+          }`}
+        >
+          {[1, 2, 3].map((elem, index) => (
+            <CardContent key={index}>
```
The Degree Plan display should be encapsulated in its own component for re-usability. I would suggest making the new component take in an array of Courses as its props so that it can be used for both Home Page and Main Page.

lokeshyerneni replied:
Resolved temporarily

lokeshyerneni committed:
feat(landing page): Complete landing page

Landing page has full functionality and sbould not have any bugs

fix #109

lokeshyerneni committed:
fix(landing page): Fixed Caleb's PR suggestions

fix #128

ZzRanger committed:
chore: Format code using Prettier

Previously, there was a new prettier config file that caused all the code to be
formatted with double quotes as opposed to single  quotes. This commit reverts
that.

ZzRanger committed:
fix: Fix build error

ZzRanger committed:
chore(home): Fix css for TopAndSidebar component

ZzRanger committed:
feat(home): Implement degree screen on home page

ZzRanger committed:
feat(home): Implement profile screen on home page

ZzRanger committed:
feat(home): Implement credits page for home

ZzRanger committed:
chore(home): Integrate TopAndSidebar component into home page

ZzRanger committed:
chore(home): Change color of home page text to white

ZzRanger committed:
chore: Remove uninformative alert

ZzRanger committed:
fix: Fix build error

ZzRanger committed:
Merge branch 'develop' into feature/home-pages

ZzRanger committed:
chore: Update package.json and fix fugs in degree planning tool

Reverted to .json data for courses for degree planning tool. Also fixed button
color issue

ZzRanger committed:
chore: Fix build errors in home pages

ZzRanger committed:
Merge branch 'feature/home-pages' into chore/merge-everything

ZzRanger committed:
Merge branch 'chore/merge-everything' into feature/landing-page

ZzRanger committed:
chore: Update auth routes

ZzRanger committed:
chore: Fix build errors

ZzRanger commented:
Closed from #265
